framework/src/main/java/org/checkerframework/framework/flow/CFCFGBuilder.java

```diff
@@ -159,27 +153,29 @@ protected VariableTree createEnhancedForLoopIteratorVariable(
         @Override
         protected VariableTree createEnhancedForLoopArrayVariable(
                 ExpressionTree expression, VariableElement variableElement) {
-            // We do not want to cache flow-insensitive types
-            // retrieved during CFG building.
-            boolean oldShouldCache = factory.shouldCache;
-            factory.shouldCache = false;
-            AnnotatedTypeMirror annotatedArrayType = factory.getAnnotatedType(expression);
-            factory.shouldCache = oldShouldCache;
-            if (annotatedArrayType.getKind() == TypeKind.WILDCARD
-                    && ((AnnotatedWildcardType) annotatedArrayType).isUninferredTypeArgument()) {
-                TypeMirror type = TreeUtils.typeOf(expression);
-                AnnotatedArrayType newArrayType =
-                        (AnnotatedArrayType) AnnotatedTypeMirror.createType(type, factory, false);
-                newArrayType.setComponentType(annotatedArrayType);
-                newArrayType.addAnnotations(annotatedArrayType.getEffectiveAnnotations());
-                annotatedArrayType = newArrayType;
+            // It is necessary to get the elt because just getting the type of expression
+            // directly (via TreeUtils.typeOf) doesn't include annotations on the declarations
+            // of local variables, for some reason.
+            Element elt = TreeUtils.elementFromTree(expression);
+            TypeMirror type = null;
+            if (elt != null) {
+                type = ElementUtils.getType(elt);
```

Comment:
This isn't the correct type.  It hasn't been view-point adapted to the use of the element.  If it's only local variables whose annotations are missing, then you can check if expression is a use of a local variable and use the element type in that case.

Reply:
> It hasn't been view-point adapted to the use of the element.

Yes, I'm aware. We had to make a [small change in Daikon](https://github.com/codespecs/daikon/pull/312) to accommodate this change, but Mike and I decided that it was an acceptable price to pay. #4189 requires this code not to rely on a particular annotated type factory being available, which is why this can't be view-point adapated.

That said, I've changed this code so that this path is only taken for local variables.